### PR TITLE
WIP: CI: Test rebuild all stacks

### PR DESCRIPTION
**Pull request by kotfic**
This is a continuation of https://github.com/spack/spack/pull/29281

### CODE REVIEW

File: share/spack/gitlab/cloud_pipelines/stacks/build_systems/spack.yaml [hunk 1]
```diff
@@ -36,12 +36,57 @@ spack:
...
+          - xz
+          - zlib
+        runner-attributes:
+           variables:
+             CI_JOB_SIZE: "small"
+             KUBERNETES_CPU_REQUEST: "500m"
+             KUBERNETES_MEMORY_REQUEST: "500M"
+      - match:
+          - 'os=ubuntu18.04'
+        runner-attributes: {}
```
Comment (scottwittenburg) [inline share/spack/gitlab/cloud_pipelines/stacks/build_systems/spack.yaml:89]: You don't want to specify some variables for the "catch-all" match?

File: share/spack/gitlab/cloud_pipelines/stacks/build_systems/spack.yaml [hunk 2]
```diff
@@ -36,12 +36,57 @@ spack:
...
+    image:
+      name: "ghcr.io/spack/e4s-ubuntu-18.04:v2021-10-18"
+      entrypoint: [ "" ]
+    tags:
+      - glr-testing
 
     mappings:
-      - match: [ 'os=ubuntu18.04' ]
+      - match:
+          - cmake
```
Comment (scottwittenburg) [inline share/spack/gitlab/cloud_pipelines/stacks/build_systems/spack.yaml:47]: I'm not sure `cmake` needs the huge, I think so far we've let it land in the catch-all, which is usually mapped to the smallest instance type.

**Review by scottwittenburg — COMMENTED** [review]
Couple comments and let's see how the build_systems pipeline looks.

File: share/spack/gitlab/cloud_pipelines/stacks/build_systems/spack.yaml (hunk 1 above)
Comment (kotfic) [reply share/spack/gitlab/cloud_pipelines/stacks/build_systems/spack.yaml:89]: this will be handled by the underlying runner default configuration

**Pull request closed without merging**